### PR TITLE
Add compile-test

### PR DESCRIPTION
**Pull request by messense**
#34

### CODE REVIEW

File: tests/compile-fail/modinit-used-in-non-fn.rs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#![feature(proc_macro)]
+extern crate pyo3;
+
+use pyo3::{py, PyResult, Python, PyModule};
+
+#[py::modinit(rust2py)]  //~ ERROR: custom attribute panicked
```
Comment (messense) [inline tests/compile-fail/modinit-used-in-non-fn.rs]: Tried to use `//~ ERROR: #[modinit] can only be used with fn block` but did not work.
Are we doing something wrong when emitting errors from proc macro?

**Comment by messense** [comment]
Travis CI service seems down.

**Comment by fafhrd91** [comment]
This tests are fine, but I want test gil behavior, we should have static guarantees that gil usage is right

**Comment by messense** [comment]
Sure, definitely should add GIL tests.

**Pull request merged**